From 167f5bfbe13f8cb5c2281b3c5bc126ebd86b72d1 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Sun, 1 Mar 2026 18:14:02 +0100
Subject: [PATCH 53/75] COMP: Expose extension build vars in install-tree
 SlicerConfig

---
 CMake/SlicerInstallConfig.cmake.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMake/SlicerInstallConfig.cmake.in b/CMake/SlicerInstallConfig.cmake.in
index dde5a14b4c..08c1a22d24 100644
--- a/CMake/SlicerInstallConfig.cmake.in
+++ b/CMake/SlicerInstallConfig.cmake.in
@@ -184,6 +184,9 @@ if(Slicer_USE_PYTHONQT)
   set(Slicer_INSTALL_QTSCRIPTEDMODULES_SHARE_DIR "@Slicer_INSTALL_QTSCRIPTEDMODULES_SHARE_DIR@")
 endif()
 
+# Extension build support (required when building extensions against an installed Slicer)
+set(Slicer_EXTENSION_GENERATE_CONFIG "${Slicer_CMAKE_DIR}/SlicerExtensionGenerateConfig.cmake")
+
 # Components needed to build modules from outside a Slicer build tree or
 # from outside a Slicer installation tree
 # set(TCLAP_DIR "${Slicer_HOME}/lib/tclap")
-- 
2.52.0

