diff --git a/Modules/Loadable/Reformat/CMakeLists.txt b/Modules/Loadable/Reformat/CMakeLists.txt
index 97423fe59c..e7ff9d5831 100644
--- a/Modules/Loadable/Reformat/CMakeLists.txt
+++ b/Modules/Loadable/Reformat/CMakeLists.txt
@@ -5,6 +5,10 @@ project(SlicerReformat)
 find_package(Slicer REQUIRED)
 include(${Slicer_USE_FILE})
 
+# Ensure CMake knows where the macro lives; try the installed CMake folder first
+include("${Slicer_DIR}/CMake/SlicerMacroBuildModuleLogic.cmake")
+include("${Slicer_DIR}/CMake/SlicerMacroBuildModuleVTKLibrary.cmake")
+
 #-----------------------------------------------------------------------------
 # Extension meta-information
 set(EXTENSION_HOMEPAGE "https://github.com/ALive-research/Slicer-Liver")
