From 48e103fb8c52807fc5d51136546e15300e81b2f4 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Fri, 27 Feb 2026 16:31:41 +0100
Subject: [PATCH] COMP: Add qMRMLWidgets to MODULE_TARGET_LIBRARIES for simple
 modules

---
 CreateModels/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CreateModels/CMakeLists.txt b/CreateModels/CMakeLists.txt
index 52eceb8..1eab35e 100644
--- a/CreateModels/CMakeLists.txt
+++ b/CreateModels/CMakeLists.txt
@@ -53,6 +53,7 @@ set(MODULE_UI_SRCS
 
 set(MODULE_TARGET_LIBRARIES
   vtkSlicer${MODULE_NAME}ModuleLogic
+  qMRMLWidgets
   )
 
 set(MODULE_RESOURCES
-- 
2.52.0

