From 29fc459cccbf0d40225a544dad782587a62abb23 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Thu, 9 Apr 2026 09:26:38 +0200
Subject: [PATCH] ENH: Add standalone build support for CropVolumeSequence
 scripted module 5.10

---
 .../Scripted/CropVolumeSequence/CMakeLists.txt  | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/Modules/Scripted/CropVolumeSequence/CMakeLists.txt b/Modules/Scripted/CropVolumeSequence/CMakeLists.txt
index adf25235ba..53d9bebc51 100644
--- a/Modules/Scripted/CropVolumeSequence/CMakeLists.txt
+++ b/Modules/Scripted/CropVolumeSequence/CMakeLists.txt
@@ -1,3 +1,20 @@
+cmake_minimum_required(VERSION 3.16.3...3.19.7 FATAL_ERROR)
+project(CropVolumeSequence)
+
+set(CMAKE_CXX_STANDARD "17")
+
+find_package(Slicer REQUIRED)
+include(${Slicer_USE_FILE})
+
+if(NOT Slicer_QTSCRIPTEDMODULES_LIB_DIR)
+  set(Slicer_QTSCRIPTEDMODULES_LIB_DIR
+      "lib/Slicer-${Slicer_VERSION}/qt-scripted-modules")
+endif()
+if(NOT Slicer_INSTALL_QTSCRIPTEDMODULES_LIB_DIR)
+  set(Slicer_INSTALL_QTSCRIPTEDMODULES_LIB_DIR
+      "lib/Slicer-${Slicer_VERSION}/qt-scripted-modules")
+endif()
+
 #-----------------------------------------------------------------------------
 set(MODULE_NAME CropVolumeSequence)
 
-- 
2.52.0

