diff --git a/CMake/SlicerBlockInstallTBB.cmake b/CMake/SlicerBlockInstallTBB.cmake
index 0c118685b4..70156cd703 100644
--- a/CMake/SlicerBlockInstallTBB.cmake
+++ b/CMake/SlicerBlockInstallTBB.cmake
@@ -4,12 +4,12 @@
 
 set(TBB_INSTALL_LIB_DIR "${Slicer_INSTALL_LIB_DIR}")
 
-if(NOT EXISTS "${TBB_BIN_DIR}")
-  message(FATAL_ERROR "CMake variable TBB_BIN_DIR is set to a nonexistent directory: ${TBB_BIN_DIR}")
-endif()
-if(NOT EXISTS "${TBB_LIB_DIR}")
-  message(FATAL_ERROR "CMake variable TBB_LIB_DIR is set to a nonexistent directory: ${TBB_LIB_DIR}")
-endif()
+# if(NOT EXISTS "${TBB_BIN_DIR}")
+#   message(FATAL_ERROR "CMake variable TBB_BIN_DIR is set to a nonexistent directory: ${TBB_BIN_DIR}")
+# endif()
+# if(NOT EXISTS "${TBB_LIB_DIR}")
+#   message(FATAL_ERROR "CMake variable TBB_LIB_DIR is set to a nonexistent directory: ${TBB_LIB_DIR}")
+# endif()
 
 if(WIN32)
   install(
diff --git a/CMake/SlicerCPack.cmake b/CMake/SlicerCPack.cmake
index 507852cbf1..090eb0b55a 100644
--- a/CMake/SlicerCPack.cmake
+++ b/CMake/SlicerCPack.cmake
@@ -473,4 +473,7 @@ DeleteRegKey SHCTX \\\"SOFTWARE\\\\Classes\\\\${ext}\\\"
 
 endif()
 
-include(CPack)
+# Include CPack if it's not included yet
+if(CPack_CMake_INCLUDED EQUAL 1)
+    include(CPack)
+endif()
