diff --git a/CMake/SlicerBlockCXXRequiredFlags.cmake b/CMake/SlicerBlockCXXRequiredFlags.cmake
index c4fa2aacdd..53ea1d7323 100644
--- a/CMake/SlicerBlockCXXRequiredFlags.cmake
+++ b/CMake/SlicerBlockCXXRequiredFlags.cmake
@@ -36,7 +36,7 @@ if(NOT DEFINED Slicer_REQUIRED_C_FLAGS OR NOT DEFINED Slicer_REQUIRED_CXX_FLAGS)
   set(tmp_cxx_flags)
 
   if(CMAKE_COMPILER_IS_GNUCXX)
-    set(cflags "-Wall -Wextra -Wpointer-arith -Winvalid-pch -Wcast-align -Wwrite-strings -D_FORTIFY_SOURCE=2")
+    set(cflags "-Wall -Wextra -Wpointer-arith -Winvalid-pch -Wcast-align -Wwrite-strings")
     SlicerFunctionCheckCompilerFlags("-fdiagnostics-show-option" cflags)
     SlicerFunctionCheckCompilerFlags("-Wl,--no-undefined" cflags)
 
